packages/components/src/overlays/Popover/src/Popover.tsx

```diff
@@ -0,0 +1,126 @@
+    type PopoverProps as RACPopoverProps,
+    type DialogTriggerProps,
+    DialogTrigger,
+    Popover as RACPopover,
+    composeRenderProps,
+    Dialog
+} from "react-aria-components";
+
+import { PopoverContext } from "./PopoverContext.ts";
+
+import styles from "./Popover.module.css";
+
+export const GlobalPopoverCssSelector = "hop-Popover";
+
+export interface PopoverProps extends StyledComponentProps<RACPopoverProps> {
+}
+
+export interface PopoverTriggerProps extends DialogTriggerProps {
+}
+
+export const PopoverTrigger = (props: PopoverTriggerProps) =>
+    <DialogTrigger {...props}>{props.children}</DialogTrigger>;
+
+function Popover(props: PopoverProps, ref: ForwardedRef<HTMLElement>) {
+    [props, ref] = useContextProps(props, ref, PopoverContext);
+    const { stylingProps, ...ownProps } = useStyledSystem(props);
+    const {
+        children,
+        className,
+        style: styleProp,
+        offset = 4,
+        ...otherProps
+    } = ownProps;
+
+    const { colorScheme } = useColorSchemeContext();
+    
+    const popoverClassNames = composeClassnameRenderProps(
+        className,
+        GlobalPopoverCssSelector,
+        cssModule(
+            styles,
+            "hop-Popover"
+        )
+    );
+
+    const style = composeRenderProps(styleProp, prev => {
+        return {
+            ...prev
+        };
+    });
+
+    return (
+        <RACPopover
+            {...otherProps}
+            offset={offset}
+            ref={ref}
+            className={popoverClassNames}
+            style={style}
+        >
+            <HopperProvider colorScheme={colorScheme}>
+                <Dialog className={clsx(stylingProps.className, styles["hop-Popover__dialog"])}
+                    style={stylingProps.style}
+                >
+                    <SlotProvider values={[
+                        [TextContext, {
+                            className: styles["hop-Popover__title"],
+                            size: "md"
+                        }],
+                        [HeadingContext, {
+                            className: styles["hop-Popover__title"],
+                            size: "inherit"
+                        }],
+                        [ButtonContext, {
+                            size: "sm",
+                            className: styles["hop-Popover__action"]
+                        }],
+                        [ButtonGroupContext, {
+                            size: "sm",
+                            align: "end",
+                            className: styles["hop-Popover__actions"]
+                        }],
+                        [FooterContext, {
+                            className: styles["hop-Popover__footer"]
+                        }],
+                        [LinkContext, {
+                            size: "sm",
+                            isQuiet: true
+                        }]
+                    ]}
+                    >
+                        <>
```

Comment:
unnecesseray fragment

Reply:
I have to add it otherwise a TS error appears 

<img width="672" alt="Error-TS" src="https://github.com/gsoft-inc/wl-hopper/assets/2379339/7b250cd1-b8e6-4f27-b01e-1e910f4f5063">

Reply:
@alexasselin008 Is it okay to keep the fragmentation or do you think the code needs to be modified more thoroughly?

Reply:
Oh this error is important, and adding the fragment hides the issues. Children could be a function, and at the moment you don't pass de necessary values. 

```
<Dialog>
   {state => (
       <SlotProvider>
            {isFunction(children)? children(state) : children}
       </SlotProvider>
   )}
```